app/src/main/java/com/mobile/tandil/kotlinbaseproject/activity/MainActivity.kt

```diff
@@ -30,14 +31,21 @@ class MainActivity : AppCompatActivity(), ListenerSetParkingSpaces {
         binding.updateButton.setOnClickListener {
             viewModel.showParkingSizeFragment()
         }
+        binding.parkingStayButton.setOnClickListener {
+            viewModel.showParkingStayFragment()
+        }
     }
 
     private fun updateUI(data: ParkingData) {
         when (data.state) {
-            CheckState.SHOW_FRAGMENT -> {
+            CheckState.SHOW_SIZE_FRAGMENT -> {
                 val newFragment: ParkingSizeUpdateDialogFragment = ParkingSizeUpdateDialogFragment.newInstance(this@MainActivity)
                 newFragment.show(supportFragmentManager, ParkingSizeUpdateDialogFragment.MAIN_DIALOG_KEY)
             }
+            CheckState.SHOW_PARKING_STAY_ACTIVITY -> {
+                val intent = Intent(this, ParkingStayActivity::class.java)
```

Comment:
do not create a variable if you will use it only in one place. Please create and call a newIntent static method from the ParkingStayActivity to return the necessary intent